### PR TITLE
envoy: Bump envoy to v1.25.4

### PR DESCRIPTION
**Pull request by sayboras**

### CODE REVIEW

**Comment by sayboras** [comment]
/test-1.26-net-next

**Comment by sayboras** [comment]
/test-1.26-net-next

**Comment by sayboras** [comment]
ci-eks failure is due to https://github.com/cilium/cilium/issues/24774

```
2023-04-06T05:24:51.260101017Z level=info msg="Initializing IPAM" mode=eni subsys=cilium-operator-aws
2023-04-06T05:24:51.262287672Z level=info msg="Starting ENI allocator..." subsys=ipam-allocator-aws
2023-04-06T05:24:54.209325867Z level=warning msg="Unable to synchronize EC2 VPC list" error="operation error EC2: DescribeVpcs, exceeded maximum number of attempts, 3, https response error StatusCode: 0, RequestID: , request send failed, Post \"https://ec2.us-east-2.amazonaws.com/\": dial tcp: lookup ec2.us-east-2.amazonaws.com on 10.100.0.10:53: write udp 192.168.189.83:56692->10.100.0.10:53: write: operation not permitted" subsys=eni
2023-04-06T05:24:54.209357228Z level=fatal msg="Unable to start eni allocator" error="Initial synchronization with instances API failed" subsys=cilium-operator-aws
```

File: pkg/envoy/accesslog_server.go [hunk 1]
```diff
@@ -164,11 +164,11 @@ func logRecord(pblog *cilium.LogEntry) *logger.LogRecord {
 	} else {
 		// Default to the deprecated HTTP log format
```
Comment (jrajahalme) [inline pkg/envoy/accesslog_server.go]: IMO we should just drop this block altogether. If `pblog.GetHttp()` is non-nil we already take the first `if` statement above.

**Review by jrajahalme — CHANGES_REQUESTED** [review]
Better remove the stale accesslog server code block.

**Comment by sayboras** [comment]
Rebase with master to get the recent CI fixes for ci-eks and ci-datapath.

**Review by meyskens — APPROVED** [review]
LGTM

**Review by jrajahalme — APPROVED** [review]

**Comment by sayboras** [comment]
test-1.26-net-next hits https://github.com/cilium/cilium/issues/15455, the error message is exactly same.

```
 	 error: unable to upgrade connection: Authorization error (user=kube-apiserver-kubelet-client, verb=create, resource=nodes, subresource=proxy)
```

**Comment by sayboras** [comment]
/test-1.26-net-next

**Review by tklauser — APPROVED** [review]

**Pull request merged**